omneeg/io.py

```diff
@@ -56,18 +58,27 @@ def __getitem__(self, i):
                 print('File format not supported')
             if self.rename_channels:
                 mne.rename_channels(eeg.info, self.rename_channels)
+                # Set non-EEG channels to misc type
+                #TODO: There must be a more data-efficient way
+                misc_channels = ['PG1', 'PG2', 'EKG', 'SP2', 'SP1', 'RLC', 'LUC', 'CH30', 'PHOTIC']
+                for ch in misc_channels:
+                    if ch in eeg.ch_names:
+                        eeg.set_channel_types({ch: 'misc'})
             if self.set_montage:
                 montage = mne.channels.make_standard_montage(self.set_montage)
-                eeg.set_montage(montage)
+                eeg.set_montage(montage, on_missing='ignore')
             eeg.resample(sfreq=self.sfreq)
             if self.create_epochs:
                 eeg = mne.make_fixed_length_epochs(
                     eeg, duration=self.duration, preload=True)
             eeg.pick_types(meg=False, eeg=True, eog=False)
             if not os.path.exists(os.path.dirname(output)):
                 os.makedirs(os.path.dirname(output))
-            out = Interpolate((self.resolution, self.resolution))(
-                eeg[:self.epochs])
+            out = Interpolate(
+                (self.resolution, self.resolution),
+                transform_type=self.transform_type,
+                l_max=self.l_max
+            )(eeg[:self.epochs])
```

Comment:
Again, maybe more zen and pythonesque to just pass the transform type and resolution